pkg/storage/tree/serialize.go

```diff
@@ -340,19 +227,34 @@ func (t *Tree) SerializeTruncateNoDict(maxNodes int, w io.Writer) error {
 			return err
 		}
 
-		val := tn.Self
+		if _, err = vw.Write(w, tn.Self); err != nil {
+			return err
+		}
 		cNodes := tn.ChildrenNodes
 		tn.ChildrenNodes = tn.ChildrenNodes[:0]
```

Comment:
I don't fully get what we're gaining by doing this, but I left it untouched. I did not copy this to the other implementation (`SerializeTruncate`) because we don't want to modify the tree there